scaladoc/src/dotty/tools/scaladoc/renderers/Resources.scala

```diff
@@ -150,7 +150,7 @@ trait Resources(using ctx: DocContext) extends Locations, Writer:
             val entry = mkEntry(member.dri, member.name, flattenToText(sig), descr, member.kind.name)
             val children = member
                 .membersBy(m => m.kind != Kind.Package && !m.kind.isInstanceOf[Classlike])
-                .filter(m => m.origin == Origin.RegularlyDefined && m.inheritedFrom.isEmpty)
+                .filter(m => m.origin == Origin.RegularlyDefined && (m.inheritedFrom.isEmpty || m.inheritedFrom.get.isSourceSuperclassHidden))
```

Comment:
NIT:
shorter and without `get`
`m.inheritedFrom.fold(true)(_.isSourceSuperclassHidden)`